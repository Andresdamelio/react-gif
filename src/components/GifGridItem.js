import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ image }) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={image.url} alt={image.title} />
      <p className='paragraph'>{image.title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  image: PropTypes.object.isRequired
};

export default GifGridItem;
