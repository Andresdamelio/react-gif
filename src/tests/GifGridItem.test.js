import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../components/GifGridItem';

describe('Probando componente <GifGridItem />', () => {
  test('El componente <GifGridItem /> debe de mostrarse correctamente', () => {
    const image = {
      url: 'https://media1.giphy.com/media/l41JU9pUyosHzWyuQ/giphy.gif?cid=9d23b6ac5mxoxi5d87is76lys6chl2gstpg822syzg1jrb8o&rid=giphy.gif&ct=g',
      title: 'Rick And Morty GIF by Adult Swim'
    };
    const wrapper = shallow(<GifGridItem image={image} />);

    expect(wrapper).toMatchSnapshot();
  });
});
