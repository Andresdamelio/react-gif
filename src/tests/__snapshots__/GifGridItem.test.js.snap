// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifGridItem /> El componente <GifGridItem /> debe de mostrarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Rick And Morty GIF by Adult Swim"
    src="https://media1.giphy.com/media/l41JU9pUyosHzWyuQ/giphy.gif?cid=9d23b6ac5mxoxi5d87is76lys6chl2gstpg822syzg1jrb8o&rid=giphy.gif&ct=g"
  />
  <p
    className="paragraph"
  >
    Rick And Morty GIF by Adult Swim
  </p>
</div>
`;
